apps/molgenis-components/src/client/client.js

```diff
@@ -154,35 +207,64 @@ const request = async (url, graqphl) => {
   return result.data.data;
 };
 
-export { request };
+const executeSaveCommand = (graphqlURL, tableName, isDraft, value, action) => {
+  // TODO: add spinner
 
-export default {
-  newClient: (graphqlURL, externalAxios) => {
-    const myAxios = externalAxios || axios;
-    // use closure to have metaData cache private to client
-    let metaData = null;
-    return {
-      fetchMetaData: () => {
-        return fetchMetaData(myAxios, graphqlURL).then((schema) => {
-          metaData = schema;
-          // node js may not have structuredClone function, then fallback to deep clone via JSON
-          return typeof structuredClone === "function"
-            ? structuredClone(metaData)
-            : JSON.parse(JSON.stringify(metaData));
-        });
-      },
-      fetchTableData: async (tableId, properties) => {
-        if (metaData === null) {
-          metaData = await fetchMetaData(myAxios, graphqlURL);
-        }
-        return fetchTableData(
-          tableId,
-          properties,
-          metaData,
-          myAxios,
-          graphqlURL
-        );
-      },
-    };
-  },
+  value.mg_draft = isDraft;
+
+  const variables = { value: [value] };
+  const query = getUpsertQuery(tableName, action);
+  return requestMultipart(graphqlURL, query, variables);
 };
+
+function getUpsertQuery(tableName, action) {
+  const name = tableName;
+  return `mutation ${action}($value:[${name}Input]){${action}(${name}:$value){message}}`;
+}
+
+function requestMultipart(url, query, variables) {
+  return new Promise(function (resolve, reject) {
+    //thanks to https://medium.com/@danielbuechele/file-uploads-with-graphql-and-apollo-5502bbf3941e
+    const formData = new FormData();
+
+    // search for File objects on the request and set it as formData
+    for (let { node, path } of new RecursiveIterator(variables)) {
```

Comment:
is this worth adding a dependency ? ( looks like a copy from the medium post)

Reply:
Same answer ;D